### PR TITLE
First round of provider tests

### PR DESCRIPTION
**Pull request by cb-jake**
### _Summary_

Add initial tests for `CoinbaseWalletProvider`

### _How did you test your changes?_

w/ unit tests

### CODE REVIEW

File: src/CoinbaseWalletSDK.test.ts [hunk 1]
```diff
@@ -96,10 +96,15 @@ describe("CoinbaseWalletSDK", () => {
...
 
-      test("@disconnect", () => {
+      test("@disconnect", async () => {
+        jest
+          .spyOn(mockExtensionProvider, "close")
+          // @ts-expect-error expect string instead of void
+          .mockImplementation(() => "mockClose");
+        coinbaseWalletSDK2.disconnect();
         // Calls extension close
         coinbaseWalletSDK2.disconnect();
```
Comment (brianchitester) [inline src/CoinbaseWalletSDK.test.ts:106]: double disconnect

File: src/provider/CoinbaseWalletProvider.ts [hunk 2]
```diff
@@ -218,7 +245,7 @@ export class CoinbaseWalletProvider
 
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore
-  public setProviderInfo(jsonRpcUrl: string, chainId: number) {
+  public setProviderInfo(jsonRpcUrl: string, chainId?: number) {
```
Comment (brianchitester) [inline src/provider/CoinbaseWalletProvider.ts:248]: chainId isn't used?

File: src/__mocks__/provider.ts [hunk 3]
```diff
@@ -12,17 +12,15 @@ export class MockProviderClass extends CoinbaseWalletProvider {
...
   }
 
-  public close() {
-    return "mockClose";
+  public async close() {
+    return Promise.resolve();
   }
 
   // @ts-expect-error mock relay
   private async initializeRelay() {
```
Comment (brianchitester) [inline src/__mocks__/provider.ts:20]: nit, no longer async right?

**Review by erin-at-work — COMMENTED** [review]
looks good! will approve after Brian's feedback

File: src/provider/CoinbaseWalletProvider.ts (hunk 2 above)
Comment (cb-jake) [reply src/provider/CoinbaseWalletProvider.ts:248]: Yeah, IDK why we still have it there. We could probably remove this

File: src/__mocks__/provider.ts (hunk 3 above)
Comment (cb-jake) [reply src/__mocks__/provider.ts:20]: This method is now async.

**Review by erin-at-work — DISMISSED** [review]

**Review by erin-at-work — APPROVED** [review]

**Pull request merged**